frontend/class-opengraph-image.php

```diff
@@ -229,7 +238,28 @@ private function set_posts_page_image() {
 	 */
 	private function set_singular_image( $post_id = null ) {
 		if ( $post_id === null ) {
-			$post_id = get_queried_object_id();
+			$post_id = $this->get_queried_object_id();
```

Comment:
These 3 lines seem unnecessary as the same 3 lines are called right away in set_user_defined_image()

Reply:
We cannot assume that the methods aren't being called individually. Therefor, we need to execute this multiple times.